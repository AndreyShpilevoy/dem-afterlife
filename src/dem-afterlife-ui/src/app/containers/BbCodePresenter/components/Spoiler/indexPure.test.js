/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0 , react/jsx-filename-extension:0 */

import React from 'react';
import {shallow} from 'enzyme';
import {SpoilerPure} from './index';

const styles = {
    '.default-Spoiler-focus:focus': 'default-Spoiler-focus:focus',
    content: 'default-Spoiler-content',
    focus: 'default-Spoiler-focus',
    spoiler: 'default-Spoiler-spoiler',
    title: 'default-Spoiler-title'
};

describe('SpoilerPure', () => {
    it('unnamed component with correct message should match expected snapshot', () => {
        expect(shallow(<SpoilerPure styles={styles}>{'test'}</SpoilerPure>)).toMatchSnapshot();
    });

    it('named component with correct message should match expected snapshot', () => {
        expect(shallow(<SpoilerPure styles={styles} options={'name'}>{'test'}</SpoilerPure>)).toMatchSnapshot();
    });

    it('named component with correct message should match expected snapshot', () => {
        const wrapper = shallow(<SpoilerPure styles={styles} options={'name'}>{'test'}</SpoilerPure>);
        expect(wrapper.state().isOpen).toBeFalsy();
        wrapper.find(`.${styles.focus}`).first().simulate('click');
        expect(wrapper.state().isOpen).toBeTruthy();
    });
});