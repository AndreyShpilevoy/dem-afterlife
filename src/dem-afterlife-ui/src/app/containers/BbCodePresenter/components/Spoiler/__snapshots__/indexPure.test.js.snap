// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpoilerPure named component with correct message should match expected snapshot 1`] = `
<div
  className="default-Spoiler-spoiler"
>
  <div
    className="default-Spoiler-focus"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="default-Spoiler-title"
    >
      name
    </span>
  </div>
</div>
`;

exports[`SpoilerPure unnamed component with correct message should match expected snapshot 1`] = `
<div
  className="default-Spoiler-spoiler"
>
  <div
    className="default-Spoiler-focus"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="default-Spoiler-title"
    >
      <Connect(Term)
        term={
          Object {
            "id": 35,
            "value": "Spoiler",
          }
        }
      />
    </span>
  </div>
</div>
`;
